### DISCUSSION TITLE
Thorlabs sCMOS

### DISCUSSION THREAD

**Discussion by mohebial** [bonsai-rx/bonsai#2043] [Q&A] [answered]
Hi,
I am using an sCMOS camera from Thorlabs (model CS505MUP1 - Kiralux Polarization Camera) and I want to interface it with Bonsai. However, Bonsai is not recognizing the camera. I have installed the Video Design library, but I still haven't been able to resolve the issue. Do you have any thoughts or suggestions? Thank you! -Ali

**Comment by bruno-f-cruz** [comment]
Can this camera be recognized as a normal webcam in other applications?

**Comment by mohebial** [comment]
Not as a webcam, but they do have an SDK, and I could grab frames in MATLAB: https://www.thorlabs.com/software/THO/ThorCam/Programming/Scientific_Camera_Interfaces_Windows-2.1.zip

**Reply by bruno-f-cruz** [reply]
This will require its own interface package. We have tested the integration before as a proof of concept but the package never actually saw the light of day.  I will check with @glopesdev if we can make that repo public (I seem to recall that there were some potential issues with the thorlabs licenses) so people can at least try to build on top of it

**Reply by mohebial** [reply]
Thanks, Bruno!

**Reply by bruno-f-cruz** [reply, marked as answer]
Here you go: https://github.com/bonsai-rx/thorcam
Disclaimer: This was a proof-of-concept project done a couple of years ago during a summer course. Unfortunately, we lost access to the hardware and could not finish the package or maintain it in any way. Be prepared for some debugging to get it to work. On the plus side, we managed to get images from the camera, so it may be a good place to start :P.
